docker-compose.yml

```diff
@@ -0,0 +1,24 @@
+version: '3.7'
+
+services:
+  app:
+    build:
+      context: ./
+      dockerfile: Dockerfile
+    entrypoint:
+      - "npm"
+    command: ["start"]
+    ports:
+      - "3000:3000"
+    environment:
+      - HOST=0.0.0.0
+      - AWS_REGION=${AWS_REGION}
+      - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+      - AWS_SESSION_TOKEN=${AWS_SESSION_TOKEN}
+      - AWS_SECURITY_TOKEN=${AWS_SECURITY_TOKEN}
+      - PANOPTES_API_APPLICATION=${PANOPTES_API_APPLICATION}
+      - PANOPTES_API_HOST=${PANOPTES_API_HOST}
+    volumes:
+      - ${PWD}:/src
+      - /src/node_modules
```

Comment:
not blocking - isn't this similar to the above volume mount in `/src`? Do we want the local node modules mounted into the container or should we use the built ones? 

FWIW if you want to reuse the built modules from the container (from cache) you can use a named volume, it copies from the container to the empty named volume on first start, i've used this successfully to cache lib installs in the past,
+ https://github.com/zooniverse/sugar/blob/650dc5ac8fe0442d2bb3625dbbcf4642f2a88460/docker-compose.yml#L21-L22
+ https://github.com/zooniverse/sugar/blob/650dc5ac8fe0442d2bb3625dbbcf4642f2a88460/docker-compose.yml#L8

Reply:
I copied this across from PFE but I'm not 100% sure what it does.
https://github.com/zooniverse/Panoptes-Front-End/blob/63fe2e43b2ed1c795666610375b6adc3ccf282b2/docker-compose.yml#L25-L27

Reply:
I tested this out and it is the same named volume but it uses anonymous volumes instead of a named volume like in my sugar example

So the anonymous volume will copy the `/src/node_modules` data from the built image to the named volume and ensure that data is mounted at `/src/node_modules`. Without the the `${PWD}:/src` will mount my local dir `node_modules` dir and will most likely break the docker setup (missing libs etc).

https://docs.docker.com/compose/compose-file/#volumes
> In the absence of having named volumes with specified sources, Docker creates an anonymous volume for each task backing a service. Anonymous volumes do not persist after the associated containers are removed.
> If you want your data to persist, use a named volume and a volume driver that is multi-host aware, so that the data is accessible from any node. Or, set constraints on the service so that its tasks are deployed on a node that has the volume present.

both ways work, this way is useful but may create a lot of volumes that need to be cleaned up via `docker volume prune`.